.gitignore

```diff
@@ -1,4 +1,87 @@
+# =====================================
+# Linux
+# -------------------------------------
+*~
+
+# KDE directory preferences
+.directory
+
+# Linux trash folder which might appear on any partition or disk
+.Trash-*
+
+# =====================================
+# OSX
+# -------------------------------------
 .DS_Store
+.AppleDouble
+.LSOverride
+
+# Thumbnails
+._*
+
+# Files that might appear in the root of a volume
+.DocumentRevisions-V100
+.fseventsd
+.Spotlight-V100
+.TemporaryItems
+.Trashes
+.VolumeIcon.icns
+
+# Directories potentially created on remote AFP share
+.AppleDB
+.AppleDesktop
+Network Trash Folder
+Temporary Items
+.apdisk
+
+# =====================================
+# Windows OS
+# -------------------------------------
+# Windows image file caches
+Thumbs.db
+ehthumbs.db
+
+# Folder config file
+Desktop.ini
+
+# Recycle Bin used on file shares
+$RECYCLE.BIN/
+
+# Windows Installer files
+*.cab
+*.msi
+*.msm
+*.msp
+
+# Windows shortcuts
+*.lnk
+
+# =====================================
+# Logs
+# -------------------------------------
+logs
+*.log
+npm-debug.log*
+
+# =====================================
+# Node dependency directory
+# -------------------------------------
+# https://docs.npmjs.com/misc/faq#should-i-check-my-node-modules-folder-into-git
+node_modules
+
+# =====================================
+# Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
+# -------------------------------------
+.grunt
+
+# =====================================
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio
```

Comment:
A lot of these definitions are not applicable to this project (ie. this project will never have *.iml or *.cab) files. Can you please remove anything that isn't directly causing issues for you?